doc/md/tutorial-todo-gql-mutation-input.md

```diff
@@ -0,0 +1,422 @@
+---
+id: tutorial-todo-gql-mutation-input
+title: Mutation Inputs
+sidebar_label: Mutation Inputs
+---
+
+In this section, we continue the [GraphQL example](tutorial-todo-gql.md) by explaining how to extend the Ent code
+generator using Go templates and generate [input type](https://graphql.org/graphql-js/mutations-and-input-types/)
+objects for our GraphQL mutations that can be applied directly on Ent mutations.
+
+#### Clone the code (optional)
+
+The code for this tutorial is available under [github.com/a8m/ent-graphql-example](https://github.com/a8m/ent-graphql-example),
+and tagged (using Git) in each step. If you want to skip the basic setup and start with the initial version of the GraphQL
+server, you can clone the repository and run the program as follows:
+
+```console
+git clone git@github.com:a8m/ent-graphql-example.git
+cd ent-graphql-example 
+go run ./cmd/todo/
+```
+
+## Go Templates
+
+The Ent framework accepts external templates that can extend or override the default generated functionality of its 
+code generator. In the template below, we generate 2 input types (`CreateTodoInput` and `UpdateTodoInput`) for the
+GraphQL mutations, and add additional methods on the different builders to accept these objects as an input type.
+
```

Comment:
i think its a bit hard to follow template code if you don't know what we're generating. Maybe add the final result (you can comment out most of if for brevity, but help the reader understand what's going to happen.)

Reply:
In addition, I'd consider adding an example of some "hello world" code generation just to explain how the mechanism works (how entc / go templates / the template directory interact) before going into the actual GQL gen. WDYT?

Reply:
> d to follow template code if you don't know what we're generating. Maybe add the final result (you can comment out most of if for brevity, but help the reader understand what's going to happen.)

It's there. See line 96. 

> In addition, I'd consider adding an example of some "hello world" code generation just to explain how the mechanism works (how entc / go templates / the template directory interact) before going into the actual GQL gen. WDYT?

mm.. maybe point them to the template-doc on our website instead?